### ISSUE TITLE
raymath.h location on GNU/Linux

### ISSUE THREAD

**Issue by RDR8** [raysan5/raylib#304]
On develop branch using standard instructions.
  
`~/PS/build/raysan5/raylib/examples$ make PLATFORM=PLATFORM_DESKTOP`  
  
`gcc -o models/models_mesh_picking models/models_mesh_picking.c -O2 -s -Wall -std=c99 -D_DEFAULT_SOURCE -I. -IC:\raylib\raylib/release/linux -IC:\raylib\raylib/src -IC:\raylib\raylib/src/external  -I/usr/local/include/raylib -L. -LC:\raylib\raylib/release/linux -LC:\raylib\raylib/src  -lraylib -lglfw3 -lGL -lopenal -lm -lpthread -ldl -lX11 -lXrandr -lXinerama -lXi -lXxf86vm -lXcursor -DPLATFORM_DESKTOP`  

`models/models_mesh_picking.c:14:21: fatal error: raymath.h: No such file or directory
compilation terminated.
Makefile:492: recipe for target 'models/models_mesh_picking' failed
make: *** [models/models_mesh_picking] Error 1`  
  
There is a similar issue in ./examples/physics. raymath.h and physac.h are needed.  
  
Should we install raymath.h and physac.h alongside raylib.h in /usr/local/include/ or put it next to the including file?  
What about some of your other headers? Case by case?  
  
Considering some of your other projects like raygui, maybe a /usr/local/include/raysan directory is called for. Similar to krishna's use case with /usr/local/include/raylib that's already in ./examples/Makefile.

**Comment by raysan5** [comment]
Hi @RDR8!

Actually, that's a point I don't know how to manage yet... raylib sources include some header-only libraries that can be used as standalone libraries:

 - [`raymath`](https://github.com/raysan5/raylib/blob/develop/src/raymath.h): Vector2, Vector3, Matrix and Quaternion math; it could require some review as per #291 
 - [`camera`](https://github.com/raysan5/raylib/blob/develop/src/camera.h): 3D Camera system (free, orbital, 1st person, 3rd person, custom)
 - [`gestures`](https://github.com/raysan5/raylib/blob/develop/src/gestures.h): Touch gestures processing (Tap, DoubleTap, Swipe, Drag, Pinch)
 - [`easings`](https://github.com/raysan5/raylib/blob/develop/src/easings.h): Easing functions for animations (based on [Robert Penner](http://robertpenner.com/easing/) implementation)
 - [`physac`](https://github.com/victorfisac/Physac): 2D physics library (collision detection, resolution, dinamics...)
 - [`raygui`](https://github.com/raysan5/raygui): Simple IMGUI system with several controls for tools development

Libraries like `raymath.h`, `camera.h` and `gestures.h` are used internally by raylib, but they can also be used independently of raylib.

Libraries like `easings.h`, `physac.h` and `raygui.h` provide additional functionality to the library, in case it was required.

In latest raylib release, the installer just copied `raymath.h`, `easings.h`, `physac.h` in default MinGW include directory, just in case user required them... but I don't like that approach, I prefer the user to just copy the libraries they need in any case...

Well, this is an issue to think about...

**Comment by RDR8** [comment]
If you leave it as is, it's like a gnoob alert for you and, more importantly, a lesson for the gnoob. (^;

**Comment by RDR8** [comment]
Honestly, this is a kind of easter egg and could be handled with comments or documentation. I know you teach video game programming not introduction to linux programming so I'm gonna go read the docs and work with raylib directly for a while. Thinking...  
  
Close or retitle at will.

**Comment by FlashBlaze** [comment]
> In latest raylib release, the installer just copied raymath.h, easings.h, physac.h in default MinGW include directory, just in case user required them... but I don't like that approach, I prefer the user to just copy the libraries they need in any case...

I'm on Windows and downloaded raylib yesterday from itch.io. While trying out examples in physics folder, I'm getting 
`physics_movement.c:22:10: fatal error: physac.h: No such file or directory` error.
So is the solution to copy the necessary header files from `C:\raylib\raylib\src\` to the examples folder? Or am I doing something wrong?

**Comment by raysan5** [comment]
@FlashBlaze yes, just copy required extra libraries from raylib/src to your project directory.

**Comment by wbbradley** [comment]
I'd like to revisit this issue. Right now it's not possible to use Homebrew to get a feature complete version of raylib installed. The missing headers make usage with Homebrew a slightly frustrating experience since it feels broken out of the box. For wider usage (not requiring extra cloning and copying steps) it may benefit raylib to distribute these headers along with the rest of the libs. Thoughts?

Thank you!

**Comment by raysan5** [comment]
@wbbradley That's ok for me, I didn't configured Homebrew package... I don't know how it works...

**Comment by wbbradley** [comment]
Sounds good! I'll follow up on those threads, then...

**Comment by rfaile313** [comment]
I was looking into this today as well. I hope you do end up following up on the homebre installation @wbbradley -- for the most part, raylib works **extremely well** out of the box with homebrew, especially when you consider it against the other setup options on MacOS or on Windows. You're right in that some header files might be missing. I've found adding any missing header files to the project directory to be a quick and painless fix, but it would be nice if it didn't have to be that way.... though I say again that it is an easier setup than anything else I've experienced with Raylib across Mac and Windows (haven't built on Debian yet)
